lib/_cookieUtils.ts

```diff
@@ -49,6 +50,12 @@ export function setUserToken(userToken: string | number): void {
   }
 }
 
-export function getUserToken() {
+export function getUserToken(
+  options?: any,
+  callback?: (err: any, userToken: string) => void
+): string {
+  if (isFunction(callback)) {
+    callback(null, this._userToken);
```

Comment:
do you mean having this ?

```ts
if (isFunction(callback)) {
    callback(null, this._userToken);
    return;
}
return this._userToken
```

Reply:
Yes.

Reply:
I think it's common pattern to return undefined if callback provided or the result otherwise, but I wasn't sure how to type it simply, given the signature is also used here: https://github.com/algolia/search-insights.js/blob/d7599caa40ecfaa3a7904904fd5434e8ee26322e/lib/insights.ts#L93-L96
Then I thought always returning the result was not so bad 😅 and simple too. Any thought?

Reply:
You have to use the overload, but it's not that important. It's working.